cmd/tsbs_generate_queries/databases/mongo/devops-naive.go

```diff
@@ -102,55 +107,52 @@ func (d *NaiveDevops) GroupByTimeAndPrimaryTag(qi query.Query, numMetrics int) {
 	interval := d.Interval.MustRandWindow(devops.DoubleGroupByDuration)
 	metrics, err := devops.GetCPUMetricsSlice(numMetrics)
 	panicIfErr(err)
-	bucketNano := time.Hour.Nanoseconds()
 
 	pipelineQuery := []bson.M{
 		{
 			"$match": bson.M{
 				"measurement": "cpu",
-				"timestamp_ns": bson.M{
-					"$gte": interval.StartUnixNano(),
-					"$lt":  interval.EndUnixNano(),
+				"time": bson.M{
+					"$gte": interval.Start(),
+					"$lt":  interval.End(),
 				},
 			},
 		},
 		{
 			"$project": bson.M{
 				"_id": 0,
 				"time_bucket": bson.M{
-					"$subtract": []interface{}{
-						"$timestamp_ns",
-						bson.M{"$mod": []interface{}{"$timestamp_ns", bucketNano}},
+					"$dateFromParts": bson.M{
+						"year":  bson.M{"$year": "$time"},
+						"month": bson.M{"$month": "$time"},
+						"day":   bson.M{"$dayOfMonth": "$time"},
+						"hour":  bson.M{"$hour": "$time"},
```

Comment:
Here I used `$dateFromParts` to construct the Date instead of `$dateSubtract` as I did earlier, since we bucket with hour granularity instead of minute, and the alternative would require two nested `$dateSubtract` expressions.